NeuroID/NeuroIDClass/Extensions/NIDSession.swift

```diff
@@ -130,62 +132,96 @@ public extension NeuroID {
+            var startStatus: SessionStartResult
+
+            // The following events have to happen for either
+            //  an existing session that begins a new flow OR
+            //  a new session with a new flow
+            // 1. Determine if flow should be sampled
+            // 2. CREATE_SESSION and MOBILE_METADATA events captured
+            // 3. Capture ADV (based on global config and lib installed)
+
+            if !NeuroID._isSDKStarted {
+                // if userID passed then startSession should be used
+                if userID != nil {
+                    startStatus = NeuroID.startSession(userID)
+                } else {
+                    let started = NeuroID.start()
+                    startStatus = SessionStartResult(started, NeuroID.getUserID())
+                }
+
+                if !startStatus.started {
+                    completion(startStatus)
+                    return
+                }
 
-            captureMobileMetadata()
+            } else {
+                NeuroID.determineIsSessionSampled()
 
-            checkThenCaptureAdvancedDevice()
-        }
+                // capture CREATE_SESSION and METADATA events for new flow
+                saveEventToLocalDataStore(createNIDSessionEvent())
+                captureMobileMetadata()
 
-        if !startStatus.started {
-            return startStatus
-        }
+                checkThenCaptureAdvancedDevice()
+
+                startStatus = SessionStartResult(true, NeuroID.getUserID())
+            }
 
-        // add linkedSite var
-        NeuroID.linkedSiteID = siteID
+            NeuroID.linkedSiteID = siteID
 
-        let setLinkedSiteIDEvent = NIDEvent(sessionEvent: NIDSessionEventName.setLinkedSite)
-        setLinkedSiteIDEvent.v = siteID
-        saveEventToLocalDataStore(setLinkedSiteIDEvent)
+            // Add the SET_LINKED_SITE event for MIHR purposes
+            //  this event is ignore by the collector service
+            let setLinkedSiteIDEvent = NIDEvent(sessionEvent: NIDSessionEventName.setLinkedSite)
+            setLinkedSiteIDEvent.v = siteID
+            saveEventToLocalDataStore(setLinkedSiteIDEvent)
 
-        return startStatus
+            completion(startStatus)
+        }
     }
 }
 
 extension NeuroID {
-    static func validateSiteID(_ string: String) -> Bool {
-        let regex = #"^form_[a-zA-Z0-9]{5}\d{3}$"#
-        let predicate = NSPredicate(format: "SELF MATCHES %@", regex)
-
-        let valid = predicate.evaluate(with: string)
+    /*
+      Determine if the session/flow should be sampled (i.e. events captured and sent)
+       if not then change the _isSessionSampled var
+       this var will be used in the DataStore.cleanAndStoreEvent method
+       and will drop events if false
+     */
+    static func determineIsSessionSampled() {
+        if NeuroID.configService.configCache.currentSampleRate >= 100 {
+            NeuroID._isSessionSampled = true
```

Comment:
If sampling is > 100, then we should receive all traffic.. right?

Reply:
Correct, there was a slip through where if it equaled 100 then it wouldn't collect. But if I put <= then if it was 0 it would still collect if the rolled value was 0. This is just to double check